app/src/main/java/com/example/mob_lab1/LoginActivity.java

```diff
@@ -0,0 +1,124 @@
+package com.example.mob_lab1;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class LoginActivity extends AppCompatActivity {
+
+    EditText emailID;
+    EditText password;
+    Button btnSignUp;
+    TextView tvSignIn;
+    FirebaseAuth mFirebaseAuth;
+    private int passMinLenght = 8;
+    public String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
+
+    private FirebaseAuth.AuthStateListener mAuthStateListener;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+
+        mFirebaseAuth = FirebaseAuth.getInstance();
+        emailID = findViewById(R.id.editText);
+        password = findViewById(R.id.editText3);
+        btnSignUp = findViewById(R.id.button1);
+        tvSignIn = findViewById(R.id.textView);
+
+        mAuthStateListener = new FirebaseAuth.AuthStateListener() {
+            @Override
+            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                FirebaseUser mFirebaseUser = mFirebaseAuth.getCurrentUser();
+                if(mFirebaseUser != null){

```

Comment:
таке форматування має бути у всіх if'ів
```suggestion
                if (mFirebaseUser != null) {
```